### ISSUE TITLE
Update to 1.6.0 fails with extension version mismatch

### ISSUE THREAD

**Issue by mkindahl** [timescale/timescaledb#2513]
**Relevant system information:**
 - OS: Ubuntu 19.10, but this applies to any OS
 - PostgreSQL version (output of `postgres --version`): any
 - TimescaleDB version (output of `\dx` in `psql`): 1.6.0 and later
 - Installation method: any (but I use source)

**Describe the bug**
Upgrading to 1.6.0 from previous versions fail if you have a drop chunks-policy with `cascade_to_materialization = false`. 

**To Reproduce**
1. Build and install TimescaleDB 1.5.1
2. Create a drop chunks-policy with `cascade_to_materialization = false`.
3. Build and install TimescaleDB 1.6.0
4. Run update

**Expected behavior**
Update should succeed.

**Actual behavior**

```
postgres=# ALTER EXTENSION timescaledb UPDATE;
FATAL:  extension "timescaledb" version mismatch: shared library version 1.5.1; SQL version 1.6.1
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
```

**Comment by mkindahl** [comment]
A workaround for this issue is to drop the policies, do the update, and then re-create the policies.

**Comment by mkindahl** [comment]
Fixed by #2502, for some reason not automatically closed, so closing now.
